core/src/main/java/org/stellar/anchor/util/NetUtil.java

```diff
@@ -7,25 +7,44 @@
 import java.net.InetAddress;
 import java.net.MalformedURLException;
 import java.net.URL;
-import java.util.Objects;
 import okhttp3.Call;
 import okhttp3.Request;
 import okhttp3.Response;
 
 public class NetUtil {
+
+  /**
+   * Fetches the content from the specified URL using an HTTP GET request.
+   *
+   * <p>This method expects a response body to be present in the HTTP response. If the response is
+   * unsuccessful (i.e., not a 2xx status code) or if the response body is null, an IOException will
+   * be thrown.
+   *
+   * @param url The URL to fetch content from.
+   * @return The content of the response body as a string.
+   * @throws IOException If the response is unsuccessful, or if the response body is null.
+   */
   public static String fetch(String url) throws IOException {
+
     Request request = OkHttpUtil.buildGetRequest(url);
     Response response = getCall(request).execute();
 
-    // Check if response was successful (status code 200)
+    // Check if response was unsuccessful (ie not status code 2xx) and throw IOException
     if (!response.isSuccessful()) {
-      throw new IOException(String.format("Unable to fetch data from %s", url));
```

Comment:
Nit: It might be difficult to extract the code from a String. It might be better to introduce a custom Exception class that represents unsuccessful responses and include the code and body as a field. I believe we are only using `NetUtil` in a couple of places in the code though so we don't really need to block on this IMO.

Reply:
thanks.  i will file a ticket for the custom exception class due to time constraints and as well jakes comment about return value.  can accomodate and improve this together.

Reply:
recapturing his comment here
JakeUrban2 hours ago
Are we sure don't want to return 4xx or 5xx response bodies?

If this function only has one reference at the moment, I would suggest changing the return value to include both the response body and status code, that way callers can decide for themselves if they're interested in non-success response bodies.